core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -104,7 +104,10 @@ object ConsoleConsumer extends Logging {
     Runtime.getRuntime.addShutdownHook(new Thread() {
       override def run() {
         consumer.stop()
-
+        // if we generated a random group id (as none specified explicitly) then avoid polluting zookeeper with persistent group data, this is a hack
+        if (!conf.groupIdPassed && conf.options.has(conf.zkConnectOpt))
+          ZkUtils.maybeDeletePath(conf.options.valueOf(conf.zkConnectOpt), "/consumers/" + conf.consumerProps.get("group.id"))
```

Comment:
The finally block already contains this cleanup call
`consumer.cleanup()
      conf.formatter.close()
      reportRecordCount()

      // if we generated a random group id (as none specified explicitly) then avoid polluting zookeeper with persistent group data, this is a hack
      if (!conf.groupIdPassed)
        ZkUtils.maybeDeletePath(conf.options.valueOf(conf.zkConnectOpt), "/consumers/" + conf.consumerProps.get("group.id"))

      shutdownLatch.countDown()`

ZkUtils.maybeDeletePath is idempotent, which can be called many times without any negative impact.  I call it from the shutdown hook thread to make sure this thread will try to delete the zkNode after the main thread fails to do this.

As for the question whether we should use deleteConsumerGroupInZK to replace ZkUtils.maybeDeletePath, I recommend we use the original one since only adding two lines in the shutdown hook thread is enough to fix the bug, we do not have to do any other regression test to make sure the new deleteConsumerGroupInZK  behaves as expected. But if you insist, I will close this PR and commit a new one with AdminUtils.deleteConsumerGroupInZK.
@hachikuji , what do you think?